loopchain/peer/block_manager.py

```diff
@@ -254,16 +254,37 @@ def confirm_prev_block(self, current_block: Block):
         # start new epoch
         self.epoch = Epoch.new_epoch()
 
+    def __validate_duplication_unconfirmed_block(self, unconfirmed_block: Block):
+        last_unconfirmed_block: Block =  self.__blockchain.last_unconfirmed_block
+        try:
+            candidate_block = self.candidate_blocks.blocks[unconfirmed_block.header.hash].block
+        except KeyError:
+            # When an unconfirmed block confirmed previous block, the block become last unconfirmed block,
+            # But if the block is failed to verify, the block doesn't be added into candidate block.
+            candidate_block: Block = last_unconfirmed_block
+
+        if candidate_block is None or unconfirmed_block.header.hash != candidate_block.header.hash:
+            return
+
+        if self.__channel_service.state_machine.state == 'LeaderComplain' \
+                and self.epoch.leader_id == unconfirmed_block.header.peer_id.hex_hx():
+            raise AddUnconfirmedBlock(f"Unconfirmed block is made by complained leader. {block})")
```

Comment:
`AddUnconfirmedBlock` is not easy to guess as a name of exception for me. How do you think about it ?

Reply:
AddUnconfirmedBlock already exists we will improve exception names and use

Reply:
Change the exception name to InvalidUnconfirmedBlock